fedimint-core/src/util/mod.rs

```diff
@@ -253,19 +229,9 @@ impl Debug for SafeUrl {
     }
 }
 
-/// Only ease conversions from unsafe into safe version.
-/// We want to protect leakage of sensitive credentials unless code explicitly
-/// calls `to_unsafe()`.
-impl TryFrom<Url> for SafeUrl {
-    type Error = anyhow::Error;
-    fn try_from(u: Url) -> anyhow::Result<Self> {
-        let s = Self(u);
-
-        if s.port_or_known_default().is_none() {
```

Comment:
So the existing code looks OK ...

The port is only required if the protocol in the URL does not have have a default. `port_or_known_default`.

Otherwise ... how are we supposed to know which port to use?